src/test/java/com/google/cloud/tools/maven/run/AbstractRunMojoTest.java

```diff
@@ -0,0 +1,68 @@
+package com.google.cloud.tools.maven.run;
+
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.lang.reflect.Field;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import org.apache.maven.model.Build;
+import org.apache.maven.project.MavenProject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AbstractRunMojoTest {
+  private static class AbstractRunMojoImpl extends AbstractRunMojo {
+    @Override
+    public void execute() {
+      // do nothing;
+    }
+
+    public void injectServices(List<File> services)
+        throws NoSuchFieldException, IllegalAccessException {
+      Field servicesField = AbstractRunMojo.class.getDeclaredField("services");
+      servicesField.setAccessible(true);
+      servicesField.set(this, services);
+    }
+  }
+
+  @Mock MavenProject mavenProject;
+
+  @InjectMocks private final AbstractRunMojoImpl testMojo = new AbstractRunMojoImpl();
```

Comment:
Does `@InjectMocks` do something? (I have never used `@InjectMocks`, so I don't know what it is for.) Looks like you do manual injection with `inejctServices()`?

Reply:
Inject mocks, injects the `mavenProject` into the object at mock construction time. It doesn't work for Services because those change per test. (I could probably also do this with a package private setter)